includes/classes/Feature/VectorEmbeddings/Settings.php

```diff
@@ -0,0 +1,507 @@
+
+		$return = [];
+
+		foreach ( $post_types as $post_type ) {
+			$post_type_object  = get_post_type_object( $post_type );
+			$object_taxonomies = get_object_taxonomies( $post_type, 'objects' );
+
+			// only use public taxonomies.
+			$public_taxonomies = array_filter(
+				$object_taxonomies,
+				function ( $taxonomy ) {
+					return $taxonomy->public && 'post_format' !== $taxonomy->name;
+				}
+			);
+
+			$public_taxonomies = array_map(
+				function ( $taxonomy ) {
+					return [
+						'name'         => $taxonomy->name,
+						'label'        => $taxonomy->label,
+						'termsInclude' => [],
+						'termsExclude' => [],
+						'enabled'      => false,
+					];
+				},
+				$public_taxonomies
+			);
+
+			// this is the shape of a post type configuration.
+			$return[] = [
+				'embeddable'            => false, // whether to allow embeddings for this post type.
+				'embeddingMode'         => 'automatic', // Whether to use auto or manual embedding.
+				'enablefieldsIndexing'  => false, // Whether to flagging content inclusion via post meta.
+				'fieldsIndexingInclude' => [], // Meta fields used to flag content for inclusion.
+				'fieldsIndexingExclude' => [], // Meta fields used to flag content for exclusion. A post with an exluded
+				'fieldsEmbedding'       => [], // Fields to use for embedding generation.
+				'label'                 => $post_type_object->label, // Label used for settings panel.
+				'key'                   => $post_type, // post type name used for key in the settings object.
+				'taxonomies'            => $public_taxonomies, // Taxonomies to consider for vector embedding.
+			];
+		}
+
+		return $return;
+	}
+
+	/**
+	 * Get the post types that are applicable for indexing.
+	 *
+	 * @return array
+	 */
+	public function get_searchable_post_types() {
+		$search = \ElasticPress\Features::factory()->get_registered_feature( 'search' );
+
+		return $search->get_searchable_post_types();
+	}
+
+	/**
+	 * Get the current settings.
+	 *
+	 * @return array
+	 */
+	public function get_settings() {
+		return $this->current_settings;
+	}
+
+	/**
+	 * Get the post type configuration for a given post type.
+	 *
+	 * @param int $post_id The ID of the post to get the configuration for.
+	 * @return array The post type configuration.
+	 */
+	public function get_post_type_config( $post_id ) {
+		$post_type = get_post_type( $post_id );
+
+		$post_type_config = array_filter(
+			$this->current_settings['postTypeConfig'],
+			function ( $config ) use ( $post_type ) {
+				return $config['key'] === $post_type;
+			}
+		);
+
+		if ( empty( $post_type_config ) ) {
+			return [];
+		}
+
+		// get the first element of the array
+		return reset( $post_type_config );
+	}
+
+	/**
+	 * Check if a post is embeddable based on taxonomy and post meta conditions.
+	 *
+	 * @param int $post_id The ID of the post to check.
+	 * @return bool True if the post is embeddable, false otherwise.
+	 */
+	public function is_embeddable( $post_id ) {
```

Comment:
What do you think about opening this method with a filter, just so developers can always force a return if they want?

Reply:
Love it. I added that filter as well a notice to users on the settings screen when the filter is used.

Reply:
Done @felipeelia  - no more changes from me